sentry-log4j2/src/main/java/io/sentry/log4j2/SentryAppender.java

```diff
@@ -0,0 +1,220 @@
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginElement;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.impl.ThrowableProxy;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** Appender for Log4j2 in charge of sending the logged events to a Sentry server. */
+@Plugin(name = "Sentry", category = "Core", elementType = "appender", printObject = true)
+public final class SentryAppender extends AbstractAppender {
+  private @Nullable String dsn;
+  private @Nullable ITransport transport;
+  private @NotNull Level minimumBreadcrumbLevel = Level.INFO;
+  private @NotNull Level minimumEventLevel = Level.ERROR;
+
+  public SentryAppender(
+      String name,
+      Filter filter,
+      @Nullable String dsn,
+      @Nullable Level minimumBreadcrumbLevel,
+      @Nullable Level minimumEventLevel,
+      final @Nullable ITransport transport) {
+    super(name, filter, null, true, null);
+    this.dsn = dsn;
+    if (minimumBreadcrumbLevel != null) {
+      this.minimumBreadcrumbLevel = minimumBreadcrumbLevel;
+    }
+    if (minimumEventLevel != null) {
+      this.minimumEventLevel = minimumEventLevel;
+    }
+    this.transport = transport;
+  }
+
+  /**
+   * Create a Sentry Appender.
+   *
+   * @param name The name of the Appender.
+   * @param filter The filter, if any, to use.
+   * @return The SentryAppender.
+   */
+  @PluginFactory
+  public static SentryAppender createAppender(
+      @PluginAttribute("name") final String name,
+      @PluginAttribute("minimumBreadcrumbLevel") final Level minimumBreadcrumbLevel,
+      @PluginAttribute("minimumEventLevel") final Level minimumEventLevel,
+      @PluginAttribute("dsn") final String dsn,
+      @PluginElement("filter") final Filter filter) {
+
+    if (name == null) {
+      LOGGER.error("No name provided for SentryAppender");
+      return null;
+    }
+    return new SentryAppender(name, filter, dsn, minimumBreadcrumbLevel, minimumEventLevel, null);
+  }
+
+  @Override
+  public void start() {
+    if (dsn != null) {
+      Sentry.init(
+          options -> {
+            options.setDsn(dsn);
+            options.setSentryClientName(BuildConfig.SENTRY_LOG4J2_SDK_NAME);
+            options.setSdkVersion(createSdkVersion(options));
+            Optional.ofNullable(transport).ifPresent(options::setTransport);
+          });
+    }
+    super.start();
+  }
+
+  @Override
+  public void append(LogEvent eventObject) {
+    if (eventObject.getLevel().isMoreSpecificThan(minimumEventLevel)) {
+      Sentry.captureEvent(createEvent(eventObject));
```

Comment:
I'd get a Hub in the ctor using the HubAdapter instead of using the Sentry static class, like you did with `sentry-spring-boot-starter`

Reply:
What would be the benefit of it considering we are not mocking it anyway?

Reply:
you could assert/verify if an event and breadcrumb have been captured correctly according to the level, it'd be more important if we were using the `hub` instead of the `transport` though

Reply:
another advantage would be to not need `await.untilAsserted` in all the tests, as we'd be testing sync calls.

Reply:
+1 one on avoiding to couple with the static type here.
Dunno the test runner mode here but if we parallelize tests classes, we'd have racey behavior where one calls Sentry.close and other capture etc.

Reply:
Ok, replaced with calling `hub`.